src/metabase/models/revision/diff.clj

```diff
@@ -22,39 +22,77 @@
-    (deferred-tru "unarchived this")
+    [:public_uuid _ nil]
+    (deferred-tru "made {0} private" identifier)
 
-    [:archived true false]
-    (deferred-tru "archived this")
+    [:public_uuid nil _]
+    (deferred-tru "made {0} public" identifier)
 
-    [:dataset false true]
-    (deferred-tru "turned this into a model")
+    [:enable_embedding false true]
+    (deferred-tru "enabled embedding")
 
-    [:dataset false false]
-    (deferred-tru "changed this from a model to a saved question")
+    [:enable_embedding true false]
+    (deferred-tru "disabled embedding")
 
-    [:display _ _]
-    (deferred-tru "changed the display from {0} to {1}" (name v1) (name v2))
+    [:parameters _ _]
+    (deferred-tru "changed the filters")
 
-    [:result_metadata _ _]
-    (deferred-tru "edited the metadata")
+    [:embedding_params _ _]
+    (deferred-tru "changed the embedding parameters")
 
-    [:dataset_query _ _]
-    (deferred-tru "modified the query")
+    [:archived _ after]
+    (if after
+      (deferred-tru "archived {0}" identifier)
+      (deferred-tru "unarchived {0}" identifier))
 
-    [:collection_id nil (coll-id :guard int?)]
-    (deferred-tru "moved {0} to {1}" identifier (t2/select-one-fn :name 'Collection coll-id))
+    [:collection_position _ _]
+    (deferred-tru "changed pin position")
 
-    [:collection_id (prev-coll-id :guard int?) (coll-id :guard int?)]
+    [:collection_id nil coll-id]
+    (deferred-tru "moved {0} to {1}" identifier (if coll-id
+                                                  (t2/select-one-fn :name 'Collection coll-id)
+                                                  (deferred-tru "Our analytics")))
+
+    [:collection_id (prev-coll-id :guard int?) coll-id]
     (deferred-tru "moved {0} from {1} to {2}"
       identifier
       (t2/select-one-fn :name 'Collection prev-coll-id)
-      (t2/select-one-fn :name 'Collection coll-id))
+      (if coll-id
+        (t2/select-one-fn :name 'Collection coll-id)
+        (deferred-tru "Our analytics")))
 
     [:visualization_settings _ _]
     (deferred-tru "changed the visualization settings")
 
+    ;;  Card specific
+    [:parameter_mappings _ _]
+    (deferred-tru "changed the filter mapping")
+
+    [:collection_preview _ after]
+    (if after
+      (deferred-tru "enabled collection review")
+      (deferred-tru "disabled collection preview"))
+
+    [:dataset_query _ _]
+    (deferred-tru "modified the query")
+
+    [:dataset false true]
+    (deferred-tru "turned this into a model")
+
+    [:dataset false false]
+    (deferred-tru "changed this from a model to a saved question")
+
+    [:display _ _]
+    (deferred-tru "changed the display from {0} to {1}" (name v1) (name v2))
+
+    [:result_metadata _ _]
+    (deferred-tru "edited the metadata")
+
+    ;;  whenever database_id, query_type, table_id changed,
+    ;; the dataset_query will changed so we don't need a description for this
+    [#{:table_id :database_id :query_type} _ _]
+    nil
+
```

Comment:
Not necessarily part of this PR but this `:else nil` still bugs me. A changed column should be required to have a `diff-string`. Are you going to address it in another PR?

Reply:
like I said in the description when changing `dataset_query`, it could also changes `table_id, database_id, query_type`(query_type is unlikely but possible).

And I think having `modified query` as the description is better than `modified query and change table_id and change database_id ...`

Reply:
another way is to exclude these keys from revision, so we don't have to worry about description. but it makes the code to record revision a bit confusing.

Reply:
Great, I like this. Still, now we can remove `:else nil` and instead throw an exception? This way the code will reflect the decision to have a description for every changed column, unless explicitly stated.

Reply:
which solution do you like?

but actually now thinking more about this I don't think we can omit "table_id" from the revision.

Imagine you start with: table_id = 1, 

then you update the query and table_id = 2,

now you want to revert to the original version, if table_id is not recorded then we will have a card with dataset_query and table_id mismatched.

Reply:
about `:else nil`, it's necessary because there are changes that are not specified there.
For example in diff-strings for Dashboard, we need to handle keys that only exist on a dashboard 
https://github.com/metabase/metabase/blob/516325a22bae6974401177b68f6028792580ca92/src/metabase/models/dashboard.clj#L256

Reply:
> which solution do you like?

I meant I like the addition of this condition:
```
    [#{:table_id :database_id :query_type} _ _]
    nil
```

> For example in diff-strings for Dashboard, we need to handle keys that only exist on a dashboard
Which keys? I'm saying if there are any columns that are not in the `excluded-columns-for-dashboard-revision` set, they should be explicitly handled here. I did a quick check and it looks like `cache_ttl` , `made_public_by_id`, and `auto_apply_filters` aren't being handled yet.

Reply:
those columns only exist on dashboard so it'll be generated in https://github.com/metabase/metabase/blob/d45c263f0baebf89ec12867cb80e6b631cea1dd3/src/metabase/models/dashboard.clj#L242

notice that generating description for dashboard will first call to the default implementation, which is the function we're fixing.